api/src/main/java/com/epam/esm/config/SpringConfig.java

```diff
@@ -0,0 +1,135 @@
+package com.epam.esm.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JSR310Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.hibernate.validator.messageinterpolation.ResourceBundleMessageInterpolator;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.MessageSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.transaction.TransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+import org.springframework.validation.beanvalidation.MessageSourceResourceBundleLocator;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import javax.sql.DataSource;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+
+@Configuration
+@ComponentScan("com.epam.esm")
+@EnableWebMvc
+@PropertySource({
+        "classpath:${spring.profiles.active}.db.properties",
+        "classpath:application.properties"})
+@EnableTransactionManagement
+public class SpringConfig implements WebMvcConfigurer {
+    private final ApplicationContext applicationContext;
+
+    @Value("${application.gifts.datasource.jdbcUrl}")
+    private String jdbcUrl;
+
+    @Value("${application.gifts.datasource.driverClassName}")
+    private String driverClassName;
+
+    @Value("${application.gifts.datasource.name}")
+    private String username;
+
+    @Value("${application.gifts.datasource.password}")
+    private String password;
+
+    @Value("${application.gifts.datasource.minimumIdle}")
+    private Integer minimumIdle;
+
+    @Value("${application.gifts.datasource.maximumPoolSize}")
```

Comment:
if you are using these fields only for DataSource creation you can avoid adding them as fields:
`@Bean
    public DataSource dataSource(
     @Value("${application.gifts.datasource.jdbcUrl}") String jdbsUrl, ....) {
        HikariConfig config = new HikariConfig();
        config.setDriverClassName(driverClassName);
        config.setJdbcUrl(jdbcUrl);
        config.setUsername(username);
        config.setPassword(password);
        config.setMinimumIdle(minimumIdle);
        config.setMaximumPoolSize(maximumPoolSize);
        return new HikariDataSource(config);
    }`